src/java/org/apache/cassandra/cql3/terms/Tuples.java

```diff
@@ -0,0 +1,251 @@
+
+import org.apache.cassandra.cql3.AssignmentTestable;
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.cql3.ColumnSpecification;
+import org.apache.cassandra.cql3.QueryOptions;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.ReversedType;
+import org.apache.cassandra.db.marshal.TupleType;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+
+import static org.apache.cassandra.cql3.statements.RequestValidations.invalidRequest;
+
+/**
+ * Static helper methods and classes for tuples.
+ */
+public final class Tuples
+{
+    private Tuples() {}
+
+    public static ColumnSpecification componentSpecOf(ColumnSpecification column, int component)
+    {
+        return new ColumnSpecification(column.ksName,
+                                       column.cfName,
+                                       new ColumnIdentifier(String.format("%s[%d]", column.name, component), true),
+                                       (getTupleType(column.type)).type(component));
+    }
+
+    public static ColumnSpecification makeReceiver(List<? extends ColumnSpecification> receivers)
+    {
+        List<AbstractType<?>> types = new ArrayList<>(receivers.size());
+        StringBuilder inName = new StringBuilder("(");
+        for (int i = 0; i < receivers.size(); i++)
+        {
+            ColumnSpecification receiver = receivers.get(i);
+            inName.append(receiver.name);
+            if (i < receivers.size() - 1)
+                inName.append(',');
+            types.add(receiver.type);
+        }
+        inName.append(')');
+
+        ColumnIdentifier identifier = new ColumnIdentifier(inName.toString(), true);
+        TupleType type = new TupleType(types);
+        return new ColumnSpecification(receivers.get(0).ksName, receivers.get(0).cfName, identifier, type);
+    }
+
+    /**
+     * A raw, literal tuple.  When prepared, this will become a Tuples.Value or Tuples.DelayedValue, depending
+     * on whether the tuple holds NonTerminals.
+     */
+    public static class Literal extends Term.Raw
+    {
+        private final List<Term.Raw> elements;
+
+        public Literal(List<Term.Raw> elements)
+        {
+            this.elements = elements;
+        }
+
+        public Term prepare(String keyspace, ColumnSpecification receiver) throws InvalidRequestException
+        {
+            // The parser cannot differentiate between a tuple with one element and a term between parenthesis.
+            // By consequence, we need to wait until we know the target type to determine which one it is.
+            if (elements.size() == 1 && !checkIfTupleType(receiver.type))
+                return elements.get(0).prepare(keyspace, receiver);
+
+            TupleType tupleType = getTupleType(receiver.type);
+
+            if (elements.size() != tupleType.size())
+                throw invalidRequest("Expected %d elements in value tuple %s, but got %d: %s",
```

Comment:
It might be better with:
```suggestion
                throw invalidRequest("Expected %d elements in value for tuple %s, but got %d: %s",
```
This would require updating the expected error messages in `TupleTypeTest` and `SelectMultiColumnRelationTest`.